pkg/query/storeset.go

```diff
@@ -215,21 +225,23 @@ func (s *StoreSet) Update(ctx context.Context) {
 			if err != nil {
 				return err
 			}
+
 			innerMtx.Lock()
-			if dupSt, ok := stores[addr]; ok {
-				level.Error(s.logger).Log("msg", "duplicated address in gossip or static store nodes.", "storeAPIAddr", addr)
-				dupSt.close()
+			defer innerMtx.Unlock()
+
+			if _, ok := stores[addr]; ok {
```

Comment:
can probably do this straight away rather than try to `s.updateStore` as we will already know if the `addr` is in `store`.

Reply:
+1 the `updateStore` method should be called `specToStore` and it shouldn't deal with maps at all

Reply:
true